src/components/Clipboard.react.js

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react'; // eslint-disable-line no-unused-vars
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faCopy, faCheckCircle} from '@fortawesome/free-regular-svg-icons';
+
+/**
+ * The Clipboard component copies text to the clipboard
+ */
+
+export default class Clipboard extends React.Component {
+    constructor(props) {
+        super(props);
+        this.copyToClipboard = this.copyToClipboard.bind(this);
+        this.copySuccess = this.copySuccess.bind(this);
+        this.getText = this.getText.bind(this);
+        this.state = {
+            copied: false,
+            hasNavigator: true,
+        };
+    }
+
+    copySuccess() {
+        const showCopiedIcon = 1000;
+        this.setState({copied: true}, () => {
+            setTimeout(() => {
+                this.setState({copied: false});
+            }, showCopiedIcon);
+        });
+    }
+
+    getText() {
+        // get the inner text.  If none, use the content of the value param
+        var text = document.getElementById(this.props.target_id).innerText;
```

Comment:
Use `let` instead of `var`. But also you should only need to call `getElementById` once ie `const target = document.getElementById(stringifiedId)`

Reply:
Included in the commit.